### ISSUE TITLE
Please remember to adjust version info

### ISSUE THREAD

**Issue by IzzySoft** [abinpaul1/GPS-Video-Logger#4]
It seems that all you releases identify as version 1.0, versionCode 1 (because of [these lines](https://github.com/abinpaul1/GPS-Video-Logger/blob/master/app/build.gradle#L10) in your `build.gradle`). This not only makes it hard for the users to tell which version they're running, Android itself wouldn't accept a newer version as "update" if `versionCode` wasn't increased.

So could you please make sure that future versions have those values set appropriately? The current one would be expected to have these values set to "1.1.2" resp. "4". Thanks!

**Comment by abinpaul1** [comment]
Thanks a lot for the information, I didn't know about that earlier. I'll make sure to update it in build.gradle from now on.

**Comment by IzzySoft** [comment]
That's great, thanks! If you leave a (closing) note here when the first fixed release is available, I'll check again if I can add it to [my repo](https://apt.izzysoft.de/fdroid) to make it easier to find and access for users. If it weren't for GMS and the Google Location package (`com.google.android.gms.places_placereport`) the app uses, it could even go directly to [F-Droid](https://f-droid.org/) – but proprietary components (GMS has no libre license) are not permitted there…

**Comment by abinpaul1** [comment]
I have done a new release with the gradle changes.
I dont actually use GMS or Google Location now. I have removed those as dependancies as well. Thanks a lot for the heads up. I'll look into publishing on F-Droid

**Comment by IzzySoft** [comment]
Thanks a lot! That's fantastic! I've checked and confirm there are no more non-free components visible – so your app should be eligible for an F-Droid listing. Be welcome to apply for it by e.g. opening a [Request For Packaging](https://gitlab.com/fdroid/rfp/-/issues)! You'll be asked to establish [Fastlane Structures](https://gitlab.com/-/snippets/1895688) in your repo here (the link goes to my Cheat-Sheet). For starters, summary + description for `en-US` will do; as you have them, you can move the screenshots inside and also add an icon. That covered by a tag would make the next integration step possible.

As promised, I've just added your app to my repo. It should become [available here](https://apt.izzysoft.de/fdroid/index/apk/com.example.gps_app) with the next sync in a few hours (around breakfast time in central Europe it should be there). If you wish to link to it e.g. from your Readme, I can offer you [this badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) for visuals :smiley:

**Comment by abinpaul1** [comment]
Great, I can see it on the repo.  .
The F-droid seems to require a bit of setup. I'll get to it in time

**Comment by IzzySoft** [comment]
Yeah, takes a bit more to get an app listed there. But we help you with it (disclosure: I'm one of the F-Droid maintainers).

**Comment by IzzySoft** [comment]
As my updater just failed with the new version: You've changed the package name. Is the old one "dead"? Because it's not possible to "rename" an app. A different package name always means a different app: existing users would have to start over as if they'd installed a new app.

So for now, I'm deactivating auto-updates here until clarified. Two approaches are possible:

* keeping the app available with its original package name via Github (and hence in my repo)
* entirely switching to the new package name, effectively deprecating the "old app". In the latter case, to not make users angry: if there are any settings or data, an export/import feature (to transport data from the old to the new app) would be a good idea.

**Comment by abinpaul1** [comment]
I really regret the inconvenience 😞 . 
I had to change the name because it seems the Play Store wont accept apps with package names starting with `com.example`. I am planning to do a release on the play store. 
Once that comes through, I could upload the signed apk here in the repo as well. So that it becomes consistent across all means of distribution.
I will try to add information to the README as well on changes to the app from the old one.

**Comment by abinpaul1** [comment]
I have added a migration plan to [FAQ Section](https://github.com/abinpaul1/GPS-Video-Logger#frequently-asked-questions). 

I initially considered doing a final release for the old app name as you suggested, unfortunately never found time for it. I was a bit sceptic about maintaining different branches and having two different releases.

I was wondering if we could just put up a MESSAGE to people who visit the IzzyDroid page regarding the same and asking them to migrate. From what I understand, updates on IzzyAndroid have to be downloaded and installed manually right? We could add the steps for migrating as well. Its just a matter of copying over all files to a new folder.

I really appreciate your help and guidance

**Comment by IzzySoft** [comment]
Thanks for the migration plan! Good to see it's possible by simply copying files.

About putting up a message: that could be amended (or prepended) to the actual description, yes. But your understanding of the updates is wrong: as with other "market apps", updates are automated (unless the user switches that off). The F-Droid client used should inform the user about available updates which they then just have to confirm/install from within the app (and even that can be automated if the device is rooted, comes with a privileged extension, or runs Android > 11).

I've just added the app with its new package name (should [show up here](https://apt.izzysoft.de/fdroid/index/apk/app.gps_video_logger) with the next sync in about 11 hours from now; find [some badges here](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) if you want to link to it from your Readme, one of the two fits perfectly next to the Google one already present, the other next to the two below it). I've also amended the description for the old package name accordingly.

PS: If you wish, I can provide you the metadata in [Fastlane format](https://gitlab.com/snippets/1895688) via a PR. That way, control over descriptions and screenshots shown for your app in my repo is handed back to you – and you also could use that for deploying your app to PlayStore (if you add the Fastlane binaries; please don't ask me about details on this step as I have no idea, I just know it's possible and many projects do so).

**Comment by abinpaul1** [comment]
I have added the badges and the lastest release. I think the description in the README.md is sufficient, thanks for letting me know about the Fastlane format. Regarding F-droid, I dont think I can spend time now to set it all up locally to publish to F-droid. So I'll just keep IzzyDroid and PlayStore as the options for downloading. 
Really appreciate all the help you provided. Thanks a lot.

**Comment by IzzySoft** [comment]
> I think the description in the README.md is sufficient, thanks for letting me know about the Fastlane format.

The difference is: Fastlane can be used to automatically update description & screenshots with F-Droid repos. The Readme can't. So if you update the Readme, you'd have to remember sending out notifications, and repo maintainers had to manually update. But if you'd update fastlane, the changes could be pulled along with the new app version.

I of course gladly keep your app in my repo (regardless of your decision concerning Fastlane). F-Droid would need Fastlane – and submission is not as complicated as you might think ("set it all up locally" suggests you refer to the official docs, describing the "most preferred way" – there are alternatives to that, should you be interested).

And as always: glad I was able to help!
